components/dash-core-components/src/components/Checklist.react.js

```diff
@@ -33,68 +33,87 @@ export default class Checklist extends Component {
                 style={style}
                 className={className}
             >
-                {options.map(option => (
-                    <label
-                        key={option.value}
-                        style={labelStyle}
-                        className={labelClassName}
-                    >
-                        <input
-                            checked={includes(option.value, value)}
-                            className={inputClassName}
-                            disabled={Boolean(option.disabled)}
-                            style={inputStyle}
-                            type="checkbox"
-                            onChange={() => {
-                                let newValue;
-                                if (includes(option.value, value)) {
-                                    newValue = without([option.value], value);
-                                } else {
-                                    newValue = append(option.value, value);
-                                }
-                                setProps({value: newValue});
-                            }}
-                        />
-                        {option.label}
-                    </label>
-                ))}
+                {options.map(option => {
+                    option =
```

Comment:
I think it's not really necessary to check the row's data structure for each record one-by-one inside the map callback as we might not want to support mixed option types. If so, it's enough to check the first element and PropTypes takes care of the rest.